lib/src/model.ts

```diff
@@ -211,6 +211,13 @@ export class Model extends EventTarget {
         );
       }
 
+      public sendResults(): void {
```

Comment:
I would prefer to call the public method `retrieveFinalResult`. Singular because only one result will be sent after calling the method. `retrieve` because, although you're telling the worker to send the final result, from the client perspective you want to `get` the result (but as it is an event triggering action I think get would be confusing).

Other than naming the rest looks awesome, thanks for contributing @Clement-mim !